tsl/src/bgw_policy/job.c

```diff
@@ -301,13 +312,31 @@ policy_retention_read_and_validate_config(Jsonb *config, PolicyRetentionData *po
 
 	object_relid = ts_hypertable_id_to_relid(policy_retention_get_hypertable_id(config), false);
 	hypertable = ts_hypertable_cache_get_cache_and_entry(object_relid, CACHE_FLAG_NONE, &hcache);
-	open_dim = get_open_dimension_for_hypertable(hypertable);
+
+	open_dim = get_open_dimension_for_hypertable(hypertable, false);
+
+	/* if dim is NULL, then it should be an INTEGER partition with no int_now function */
+	if (open_dim == NULL)
+	{
+		Oid partition_type;
+
+		open_dim = hyperspace_get_open_dimension(hypertable->space, 0);
+		partition_type = ts_dimension_get_partition_type(open_dim);
+		if (!IS_INTEGER_TYPE(partition_type))
+			ereport(ERROR,
+					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+					 errmsg("incorrect partition type %d.  Expected integer", partition_type)));
+
+		/* if there's no int_now function the boundary is considered as an INTERVAL */
+		boundary_type = INTERVALOID;
+	}
```

Comment:
This is covered via tests in the https://github.com/timescale/timescaledb/pull/6227 patch